File: docs/legacy(old PACE)/java/src/main/com/pace2020/epace/sdk/sample/SystemInspector.java
package com.pace2020.epace.sdk.sample;

import java.rmi.RemoteException;
import java.util.Arrays;
import java.util.List;

import com.pace2020.epace.sdk.util.SecuredWSDLClient;

/**
 * @author <a href="mailto:jduval@pace2020.com">jerry duval</a>
 */
public class SystemInspector extends SecuredWSDLClient
{
    public static void main( final String[] args ) throws Exception
    {
        new SystemInspector().run();
    }

    public List run() throws RemoteException
    {
        /**
         * com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[Pace Mobile]
         * com.pace2020.appbox.web.form.ValidFormInspectionTarget[Pace Mobile]
         * com.pace2020.appbox.web.form.ValidFormInspectionTarget[Pace (UI 3.0)]
         * com.pace2020.epace.web.menu.UserDefinedMenuManagerManager[eService]
         * com.pace2020.appbox.blocks.inspection.CustomizationsVerificationInspectionTarget
         * com.pace2020.epace.web.menu.UserDefinedMenuManagerManager[Pace (UI 3.0)]
         * com.pace2020.appbox.web.form.ValidFormInspectionTarget[eService]
         * com.pace2020.appbox.blocks.objectmodel.inspection.XPathFieldValueVerificationInspectionTarget
         * com.pace2020.appbox.blocks.objectmodel.inspection.XPathReportParameterVerificationInspectionTarget
         * com.pace2020.epace.web.menu.UserDefinedMenuManagerManager[Pace Mobile]
         * com.pace2020.appbox.blocks.objectmodelmetadata.XPathCalculatedFieldVerificationInspectionTarget
         * com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[eService]
         * com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[Pace (UI 3.0)]
         * com.pace2020.appbox.blocks.objectmodelmetadata.OnCreateXPathDefaultVerificationInspectionTarget
         * com.pace2020.appbox.blocks.objectmodelmetadata.XPathConditionVerificationInspectionTarget
         * com.pace2020.appbox.blocks.objectmodelmetadata.OnPersistXPathDefaultVerificationInspectionTarget
         * com.pace2020.epace.blocks.event.EventHandlerDefinitionXPathVerificationInspectionTarget
         */
        return run( "com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[ePace Mobile]" );
    }

    public List<String> run( final String target ) throws RemoteException
    {
        final List<String> keys = getInspector().inspect( target ).getString();

        System.out.println( target + " has " + keys.size() + " Messages" );
        System.out.println( Arrays.asList( keys ).toString() );

        return keys;
    }
}

File: docs/legacy(old PACE)/C#/4.0/PaceAPISample/sample/SystemInspector.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using efipaceservices;

namespace Pace_Web_Service_SDK.sample
{
    class SystemInspector
    {
        public static void Run()
        {
            SystemInspectorHttpBinding systemInspectorHttpBinding = PaceClient.getSystemInspectorHttpBinding();
            /**
              * com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[ePace Mobile]
              * com.pace2020.appbox.web.form.ValidFormInspectionTarget[ePace Mobile]
              * com.pace2020.appbox.web.form.ValidFormInspectionTarget[ePace (UI 3.0)]
              * com.pace2020.epace.web.menu.UserDefinedMenuManagerManager[eService]
              * com.pace2020.appbox.blocks.inspection.CustomizationsVerificationInspectionTarget
              * com.pace2020.epace.web.menu.UserDefinedMenuManagerManager[ePace (UI 3.0)]
              * com.pace2020.appbox.web.form.ValidFormInspectionTarget[eService]
              * com.pace2020.appbox.blocks.objectmodel.inspection.XPathFieldValueVerificationInspectionTarget
              * com.pace2020.appbox.blocks.objectmodel.inspection.XPathReportParameterVerificationInspectionTarget
              * com.pace2020.epace.web.menu.UserDefinedMenuManagerManager[ePace Mobile]
              * com.pace2020.appbox.blocks.objectmodelmetadata.XPathCalculatedFieldVerificationInspectionTarget
              * com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[eService]
              * com.pace2020.epace.web.objectcontext.UserDefinedObjectContextsManager[ePace (UI 3.0)]
              * com.pace2020.appbox.blocks.objectmodelmetadata.OnCreateXPathDefaultVerificationInspectionTarget
              * com.pace2020.appbox.blocks.objectmodelmetadata.XPathConditionVerificationInspectionTarget
              * com.pace2020.appbox.blocks.objectmodelmetadata.OnPersistXPathDefaultVerificationInspectionTarget
              * com.pace2020.epace.blocks.event.EventHandlerDefinitionXPathVerificationInspectionTarget
              */
            try
            {
                String[] keys = systemInspectorHttpBinding.inspect("com.pace2020.appbox.web.form.ValidFormInspectionTarget[ePace Mobile]");
                Console.WriteLine(keys.Length + " Messages");
            }
            catch (Exception e)
            {
                Console.WriteLine(e.Message);
            }
        }
    }
}
